### DISCUSSION TITLE
`next export` and dynamic redirection

### DISCUSSION THREAD

**Discussion by jpomykala** [vercel/next.js#23688] [Help]
Hey,
I'm moving my app from CRA to NextJS. I'm using `next build && next export` with S3 to host my app. I'm wondering, how I can use dynamic routing with NextJS and `next export`? 

So far I find out that I can make `router.push(window.location.pathname)` on custom 404.tsx page. 

Is there any other way to do it? Something less hacky? I don't want to use SSR yet :) 

Thanks

**Comment by akellbl4** [comment]
Have you checked caveats in docs? 
https://nextjs.org/docs/advanced-features/static-html-export#caveats

If you don't use `getStaticPaths` for dynamic routes that page will be generated as for example `/post/[id]/index.html` and you can use it as a dynamic page with rewriting on a proxy server in front of your S3. Otherwise, it'll be just a static page that could be reached by `/post/[id]/` URL.
